From 3041a4c5ab6abc1a104f978ad49344625f264299 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Thu, 25 May 2023 10:46:48 +0600
Subject: [PATCH 102/157] arm64: configs: vc4: enable rswitch support

- enable module for proprietary Marvel 1G PHY driver
- enable module for proprietary Marvel 2.5GPHY driver
- enable module for open source Marvel 1G PHY driver
- enable module for rswitch

Enabling both proprietary and open source Marvel 1G PHY drivers because
100 Mbps mode does not work with the open source driver yet. While both
modules are enabled, userspace has to explicitly load preferred driver,
otherwise it is not guaranteed which of the two gets used.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>

diff --git a/arch/arm64/configs/renesas-vc4_defconfig b/arch/arm64/configs/renesas-vc4_defconfig
index 4bdfa3f6e5e2..57b20bd8bf58 100644
--- a/arch/arm64/configs/renesas-vc4_defconfig
+++ b/arch/arm64/configs/renesas-vc4_defconfig
@@ -294,6 +294,7 @@ CONFIG_LAN865X=m
 # CONFIG_NET_VENDOR_RDC is not set
 # CONFIG_NET_VENDOR_REALTEK is not set
 CONFIG_RAVB=y
+CONFIG_RENESAS_ETHER_SWITCH=m
 # CONFIG_NET_VENDOR_ROCKER is not set
 # CONFIG_NET_VENDOR_SAMSUNG is not set
 # CONFIG_NET_VENDOR_SEEQ is not set
@@ -310,6 +311,9 @@ CONFIG_RAVB=y
 # CONFIG_NET_VENDOR_VIA is not set
 # CONFIG_NET_VENDOR_WIZNET is not set
 # CONFIG_NET_VENDOR_XILINX is not set
+CONFIG_MARVELL_88Q2XXX_PHY=m
+CONFIG_MARVELL_88Q2110_PHY=m
+CONFIG_MARVELL_88Q3344_ES_PHY=m
 CONFIG_MICREL_PHY=y
 # CONFIG_WLAN is not set
 CONFIG_NET_FAILOVER=y
-- 
2.25.1

